packages/core/src/services/chains/ProviderProcessor/processStreamObject.ts

```diff
@@ -0,0 +1,58 @@
+import { StreamCommonData } from '../../../events/handlers'
+import { objectToString } from '../../../helpers'
+import { AIReturn } from '../../ai'
+import { saveOrPublishProviderLogs } from './saveOrPublishProviderLogs'
+
+async function processResponse({
+  aiResult,
+  commonData,
+}: {
+  commonData: StreamCommonData
+  aiResult: Awaited<AIReturn<'object'>>
+}) {
+  const object = await aiResult.data.object
```

Comment:
what happens if this promise throws?

Reply:
Great questions. This was already this way. How we should handle this? Should we create the provider log and also generate an error?